### PR TITLE
inlined endpoints on SE can now only have network field set.

### PR DESCRIPTION
**Pull request by saiskee**
**Please provide a description of this PR:**
Endpoints on service entries can now have either a `network` or `address` set.

Fixes https://github.com/istio/istio/issues/46680. 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by kdorosh — APPROVED** [review]
lgtm, will need release notes (preferable) or correct label to skip

**Review by GregHanson — APPROVED** [review]
LGTM, but please add a release note

**Comment by GregHanson** [comment]
unrelated but it looks like the docs are out of date on this since the docs and API repo still have the `address` field listed as required:
https://github.com/istio/api/blob/master/networking/v1alpha3/workload_entry.proto#L268

File: releasenotes/notes/46693.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: networking
+issue:
+  - 46680
+
+releaseNotes:
+  - |
+    **Fixed** an issue where istio validation required address to be set on endpoints even if network is set.
```
Comment (stevenctl) [inline releasenotes/notes/46693.yaml:9]: I don't know if we should write this as bug-fix. I believe it's releasing along with the WorkloadEntry update, so this is net new behavior.

**Review by stevenctl — APPROVED** [review]

File: releasenotes/notes/46693.yaml (hunk 1 above)
Comment (stevenctl) [reply releasenotes/notes/46693.yaml:9]: https://github.com/istio/istio/blob/d8b21e02e754b6e93bf259ff7e226fafaa9c16d3/releasenotes/notes/45150.yaml

Maybe we can make it just like this one.

Comment (kdorosh) [reply releasenotes/notes/46693.yaml:8]: or perhaps we should skip it if they will be in the same release notes?

Comment (GregHanson) [reply releasenotes/notes/46693.yaml:8]: +1 to Steven's suggestion

File: pkg/config/validation/validation.go [hunk 2]
```diff
@@ -3447,7 +3447,14 @@ var ValidateServiceEntry = registerValidateFunc("ValidateServiceEntry",
 						errs = appendValidation(errs, fmt.Errorf("unix endpoint %s must not include ports", addr))
 					}
 				} else {
-					errs = appendValidation(errs, ValidateIPAddress(addr))
+					if addr == "" {
+						if endpoint.GetNetwork() == "" {
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation.go:3451]: put together with addr == ""

File: pkg/config/validation/validation.go [hunk 3]
```diff
@@ -3447,7 +3447,14 @@ var ValidateServiceEntry = registerValidateFunc("ValidateServiceEntry",
...
+							errs = appendValidation(errs, errors.New("address must be set"))
+						}
+						errs = appendWarningf(errs, "address is unset with network %s", endpoint.GetNetwork())
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation.go:3454]: why warning

**Review by hzxuzhonghu — COMMENTED** [review]
I donot compare the wle validate with validateWorkloadEntry, i would suggest we reuse the logic here

File: pkg/config/validation/validation.go (hunk 3 above)
Comment (saiskee) [reply pkg/config/validation/validation.go]: this was inline with the existing validate workload entry logic

**Review by howardjohn — APPROVED** [review]

**Comment by saiskee** [comment]
@howardjohn Thank you for the approve -- can you remove the do-not-merge label?

**Pull request merged**